src/api/c/unary.cpp

```diff
@@ -560,6 +560,42 @@ af_err af_not(af_array *out, const af_array in) {
     return AF_SUCCESS;
 }
 
+template<typename T>
+static inline af_array bitOpNot(const af_array in) {
+    af_array res = unaryOp<T, af_bitnot_t>(in);
+    return res;
+}
+
+af_err af_bitnot(af_array *out, const af_array in) {
+    try {
+        const ArrayInfo &iinfo = getInfo(in);
+        const af_dtype type    = iinfo.getType();
+
+        dim4 odims = iinfo.dims();
+
+        if (odims.ndims() == 0) {
+            return af_create_handle(out, 0, nullptr, type);
+        }
+
+        af_array res;
+        switch (type) {
+            case s32: res = bitOpNot<int>(in); break;
+            case u32: res = bitOpNot<uint>(in); break;
+            case u8: res = bitOpNot<uchar>(in); break;
+            case b8: res = bitOpNot<char>(in); break;
```

Comment:
bitwise not of b8 array is not needed since, b8 is boolean and `operator!` should do fine for that one right ?